Add base performance testcase and initial TypeConversionPerformanceTest

billschaller opened the pull request:
Paves the way for performance testing in DBAL.

Commit:
Add DbalPerformanceTestCase

Commit:
Add TypeConversionPerformanceTest

Commit:
Add exclusion for performance group to phpunit.xml.dist

beberlei commented on tests/Doctrine/Tests/DBAL/Performance/TypeConversionPerformanceTest.php:
```diff
@@ -0,0 +1,30 @@
+class TypeConversionPerformanceTest extends DbalPerformanceTestCase
+{
+    /**
+     * MAXIMUM TIME: 2 seconds
```
This is a bit misleading, because it depends on hardware used. I'd rather remove it

zeroedin-bill replied:
K

Commit:
remove MAXIMUM TIME comment

Commit:
Improve structure of performance test case.
Add DbalPerformanceTestListener to collect & report performance test timing results.

Commit:
Change DbalPerformanceTestListener to report timings in __destruct instead of endTestSuite.

deeky666 commented on tests/Doctrine/Tests/DbalPerformanceTestCase.php:
```diff
@@ -0,0 +1,72 @@
+        parent::setUp();
+
+        // Reset timing vars
+        $this->startTime = $this->lastRunTime = null;
```
Those vars will be reset on each test case anyways. The whole `setUp()` can be omitted IMO

zeroedin-bill replied:
Ah - each run of a test reinstantiates the whole testcase object? Didn't realize that...

deeky666 replied:
@zeroedin-bill yes exactly.

Commit:
remove DbalPerformanceTestCase::setUp and rename lastRunTime var to just runTime

Commit:
Make output of DbalPerformanceTestListener cleaner

Commit:
add 'seconds' to output

Commit:
remove weird whitespace

deeky666 commented:
:+1:

zeroedin-bill commented:
:beers:
